### PR TITLE
feat:adds placement policy argument to the beta modules

### PR DESCRIPTION
**Pull request by g-awmalik**
Addresses #1379

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
@g-awmalik
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_generate_modules
 The modules need to be regenerated. Please run `make_build`. 
 ```bash 
Checking submodule's files generation
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' /workspace/cluster.tf /tmp/tmp.e0CtSGnOC7/workspace/cluster.tf
331a332
> 
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' /workspace/modules/private-cluster/cluster.tf /tmp/tmp.e0CtSGnOC7/workspace/modules/private-cluster/cluster.tf
344a345
> 
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' /workspace/modules/private-cluster-update-variant/cluster.tf /tmp/tmp.e0CtSGnOC7/workspace/modules/private-cluster-update-variant/cluster.tf
435a436
> 
Error: submodule's files generation has not been run, please run the
'make build' command and commit changes
 ```

**Comment by comment-bot-dev** [comment]
@g-awmalik
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@g-awmalik
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@g-awmalik
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@g-awmalik
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@g-awmalik
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by bharathkkb** [comment]
@g-awmalik agreed we don't need a dedicated example but can we add to an [existing example](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/examples/simple_regional_private_beta/main.tf) so we can assert it via an existing test.

**Comment by comment-bot-dev** [comment]
@g-awmalik
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: autogen/main/cluster.tf.tmpl [hunk 1]
```diff
@@ -612,6 +612,15 @@ resource "google_container_node_pool" "pools" {
...
   }
 
+  {% if beta_cluster %}
+  dynamic "placement_policy" {
+    for_each = length(lookup(each.value, "placement_policy", "")) > 0  ? [each.value] : []
+    content {
+      type = lookup(placement_policy.value, "placement_policy", null)
+    }
+  }
+  {% endif %}
```
Comment (bharathkkb) [inline autogen/main/cluster.tf.tmpl:622]: It looks like the block right below this is also a beta conditional, can we merge both?

**Review by bharathkkb — APPROVED** [review]
Let's also add to the nodepool vars list https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/autogen/main/README.md?plain=1#L186

**Comment by comment-bot-dev** [comment]
@g-awmalik
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@g-awmalik
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_terraform
 Failed Terraform check. More details below. 
 ```bash 
Running terraform fmt
test/fixtures/beta_cluster/main.tf
--- old/test/fixtures/beta_cluster/main.tf
+++ new/test/fixtures/beta_cluster/main.tf
@@ -46,11 +46,11 @@
 
   node_pools = [
     {
-      name             = "pool-01"
-      machine_type     = "n2-standard-2"
-      min_count        = 1
-      max_count        = 2
-      auto_upgrade     = true
+      name         = "pool-01"
+      machine_type = "n2-standard-2"
+      min_count    = 1
+      max_count    = 2
+      auto_upgrade = true
       // Beta feature
       placement_policy = "COMPACT"
     },
Error: terraform fmt failed with exit code 3
Check the output for diffs and correct using terraform fmt <dir>
 ```

**Comment by comment-bot-dev** [comment]
@g-awmalik
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@g-awmalik
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Pull request merged**